pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -362,31 +356,48 @@ public void findEntryFailed(ManagedLedgerException exception, Object ctx) {
                     topicName, subName, timestamp, result.position);
         }
 
-        cursor.asyncResetCursor(result.position, new AsyncCallbacks.ResetCursorCallback() {
-            @Override
-            public void resetComplete(Object ctx) {
-                if (log.isDebugEnabled()) {
-                    log.debug("[{}][{}] Successfully reset subscription to timestamp {}", topicName, subName,
-                            timestamp);
+        if (isFenced.compareAndSet(false, true)) {
```

Comment:
Instead of indenting the whole code, I'd prefer : 
```java
if (!isFenced.compareAndSet(false, true)) {
   // fail
   return;
}

/// continue
```